examples/ava_tools/difficulty_from_csv.py

```diff
@@ -0,0 +1,128 @@
+    return by_vid
+
+def compute_difficulty(rows):
+    """
+    Compute difficulty metrics for a single video based on bounding boxes and labels.
+    Metrics:
+      - diversity: number of unique tracked entities (proxy for people)
+      - interactivity_avg / max: average and max faces per frame
+      - dynamics: average motion between frames (1 - IoU)
+      - occlusion: average overlap between faces
+      - audibility_sna_share: share of non-audible speaking labels
+    """
+    # group by timestamp
+    by_ts = defaultdict(list)
+    entities = set()
+    for r in rows:
+        by_ts[r['ts']].append(r)
+        entities.add(r['ent'])
+
+    # interactivity
+    faces_per_frame = [len(v) for v in by_ts.values()]
+    interactivity_avg = mean(faces_per_frame) if faces_per_frame else 0.0
+    interactivity_max = max(faces_per_frame) if faces_per_frame else 0
+
+    # diversity
+    diversity = len(entities)
+
+    # audibility: share of SPEAKING_NOT_AUDIBLE among speaking frames
+    speak = [r for r in rows if r['label'].startswith('SPEAKING')]
+    sna = [r for r in speak if 'NOT_AUDIBLE' in r['label']]
+    audibility_sna_share = (len(sna) / len(speak)) if speak else 0.0
+
+    # dynamics: mean (1 - IoU) per entity over successive frames
+    by_ent = defaultdict(list)
+    for r in rows:
+        by_ent[r['ent']].append(r)
+    dynamics_terms = []
+    for ent, seq in by_ent.items():
+        seq.sort(key=lambda d: d['ts'])
+        for i in range(1, len(seq)):
+            iou_val = iou(seq[i-1]['box'], seq[i]['box'])
+            dynamics_terms.append(1.0 - iou_val)
+    dynamics = mean(dynamics_terms) if dynamics_terms else 0.0
+
+    # occlusion: average pairwise IoU per frame when >=2 faces
+    occlusion_terms = []
+    for ts, dets in by_ts.items():
+        n = len(dets)
+        if n < 2:
+            continue
+        boxes = [d['box'] for d in dets]
+        # average over all pairs
+        s, c = 0.0, 0
+        for i in range(n):
+            for j in range(i+1, n):
+                s += iou(boxes[i], boxes[j])
+                c += 1
+        if c:
+            occlusion_terms.append(s / c)
+    occlusion = mean(occlusion_terms) if occlusion_terms else 0.0
+
+    return dict(
+        diversity=diversity,
+        interactivity_avg=interactivity_avg,
+        interactivity_max=interactivity_max,
+        audibility_sna_share=audibility_sna_share,
+        occlusion=occlusion,
+        dynamics=dynamics
+    )
+
+if __name__ == "__main__":
+    import sys, os, json, glob
+    path = sys.argv[1]  
+
+    out = {}
+
+    def process_csv(csv_path):
+        by_vid = load_ava_as(csv_path)
+        for vid, rows in by_vid.items():
+            out[vid] = compute_difficulty(rows)
+
+    if os.path.isdir(path):
+        # Process every *.csv in the folder
+        for csv_path in sorted(glob.glob(os.path.join(path, "*.csv"))):
+            process_csv(csv_path)
+
+    # Save JSON next to this script
```

Comment:
next to the script can be the default but please use argparse to make this more flexible